tools/hidl/hidlcomp.cpp

```diff
@@ -4268,7 +4268,11 @@ void EspMessageInfo::write_esp()
     //method ==> copy
     outf("\nvoid C%s::copy(C%s &from)\n{\n", name_, name_);
     if (parent)
-        outf("\tC%s::copy(*dynamic_cast<C%s*>(&from));\n", parent, parent);
```

Comment:
@ghalliday I think these gave me compiler errors when I tried to make them static_cast.  But I can investigate further.